core/server/common/src/main/java/alluxio/master/MasterProcess.java

```diff
@@ -173,22 +181,19 @@ void registerServices(GrpcServerBuilder serverBuilder,
     }
   }
 
-  /**
-   * Waits until the grpc server is ready to serve requests.
-   *
-   * @param timeoutMs how long to wait in milliseconds
-   * @return whether the grpc server became ready before the specified timeout
-   */
-  public boolean waitForGrpcServerReady(int timeoutMs) {
+  @Override
+  public boolean waitForReady(int timeoutMs) {
+    if (mLeaderSelector instanceof UFSJournalSingleMasterPrimarySelector
```

Comment:
Also the if conditions are not that straightforward to follow. You may wanna break it into more readable pieces and sacrifice simplicity, like:
```
if (mLeaderSelector instanceof UFSJournalSingleMasterPrimarySelector) {
  return waitForGrpcServerReady(timeoutMs);
} else if (mLeaderSelector instanceof UFSJournalMultiMasterPrimarySelector) {
  if (mLeaderSelector.getState() == PrimarySelector.State.PRIMARY) {
    return waitForGrpcServerReady(timeoutMs);
  }
  return mServingThread == null; // is this correct?
} ...
```
I personally would prefer the expressiveness here, and just live with duplication.

Also seems `mServingThread == null;` is used as the checking condition in many places. Can we do a double check on this assumption? You can also wrap it in something more expressive like:
```
boolean isPrimary() {
  return mServingThread == null;
}
```